### DISCUSSION TITLE
Ptolemy's "geocentric" model and a terminology sanity check

### DISCUSSION THREAD

**Discussion by skintigh** [skyfielders/python-skyfield#852] [General]
Terminology has broken my brain, can I get a sanity check? Or a crudely drawn picture?

My goals:
I want to plot a "geocentric," Ptolemy-style planetarium/"orrery", whereby "geocentric" I mean "not Copernican" and "not heliocentric" and "relating to a point on the _surface_ of the Earth where some geek is gawking at the stars."
I also want to plug in "geocentric" angles (geocentric longitude (λ)? I think?), and find the date they correspond to in ancient times. (This part is possibly finished, though perhaps not accurately.)

My questions:
1. In this software, geocentric means from the center of the Earth, which would make for poor seeing, so what I really want is topocentric?
2. Also, I need to specify "apparent." I think? I thought?
3. I want the degrees away from the sun (or the degrees away from due north, both work for me.) But if I compute separation from the sun and get 90, that is an absolute value of 90 degrees of separation, so I don't know if that's 90 east or 90 west so that doesn't seem to do me any good, is that correct?
4. But I was curious: is that separation just the X value on the horizon? Or X on the ecliptic? Or is it sqrt(X^2+Y^2) of any given system? I assume the latter.
5. Bonus question: I thought I had my program right at one point, and at that time I compared values computed from NASA ephemera to values from the Almagest, and they were off... Off by exactly -2.8495 days for every date I computed in the first couple of centuries BCE. I expected NASA's computed barycentric ephemera to drift some, is that what I'm seeing? It just seemed oddly... consistent, and precise, for an error.

**Comment by brandon-rhodes** [comment]
My answers will be brief as I only have a couple of minutes free right now, but:

1. For an orrery, I don't imagine that the difference in position would matter between the geocenter and an observer. Try both and see.
2. Similarly, `.apparent()` might not make a big enough difference to be visible at the level of an orrery. But try and see. It's more correct for what you see in the sky, but less correct if you want real relative positions in space.
3. Correct!
4. You probably want to subtract longitudes: https://rhodesmill.org/skyfield/examples.html#are-planets-separated-by-0-at-conjunction-and-180-at-opposition
5. NASA is probably using UT1. Try using that instead of UTC.
